freud/environment.pyx

```diff
@@ -659,15 +714,15 @@ cdef class EnvironmentCluster(_MatchEnv):
 
 
 cdef class EnvironmentMotifMatch(_MatchEnv):
-    r"""Find matches between local arrangements of a set of points and a provided motif.
+    r"""Find matches between local arrangements of a set of points and
+    a provided motif, as done in :cite:`Teich2019`.
+
+    A particle's environment can only match the motif if it contains the
+    same number of particles as the motif. Any environment with a
+    different number of particles than the motif will always fail to match
```

Comment:
Instead of "A particle's environment can only match the motif if it contains the same number of particles as the motif. Any environment with a different number of particles than the motif will always fail to match the motif" I would say:

"A particle's environment can only match the motif if its number of neighbors equals the number of vectors in the motif. Any environment with a different number of neighbors than the number of vectors in the motif will always fail to match the motif."

I think this is clearer, and drives home the idea that the motif that the user provides and might want to search for within the system can be any set of points or vectors. It does not necessarily have to be a neighborhood that is already calculated via cluster_environments, although maybe that will be the most common use case.

Reply:
That is fine with me! I think using "vectors" is appropriate, and the most accurate.

Reply:
The term "bond vectors" threw me off in the first paragraph. I also prefer simply "vectors".

Reply:
After reading the whole thing I again I think "neighbor" is better. A major reason is that we use `env_neighbors`, and also we don't really talk about vectors until we explain why we add zero vectors. Even though we use the word neighbor a lot, I think it is pretty clear what we refer to

Reply:
My concern with using neighbors is that for someone looking at this for the first time they will easily get lost between the discussion of `env_neighbors` vs `neighbors`. IMO the documentation for both `EnvironmentCluster` and `EnvironmentMotifMatch` would be clearest if we used the word "neighbors" exceedingly sparingly:
- Environments are defined by the vectors between a particle and its nearest neighbors
- A motif is a specific environment that we expect to observe in a system.
- When clustering:
    - The nearest neighbors used to compute each particle's environment are selected based on the `env_neighbors` argument 
    - The environment of each particle is compared to the environment of its nearest neighbors, where the set of nearest neighbors is determined based on the `neighbors` argument.
- When performing motif matching:
    -  The environment of each particle is computed using the `neighbors` argument.

While writing the above, I realized that there is another significant source of confusion. The `neighbors` argument for `EnvironmentMotifMatch` is really equivalent to the `env_neighbors` argument for `EnvironmentCluster`, right?

Reply:
> While writing the above, I realized that there is another significant source of confusion. The `neighbors` argument for `EnvironmentMotifMatch` is really equivalent to the `env_neighbors` argument for `EnvironmentCluster`, right?

Yes, and I agree that our use of `env_neighbor` and `neighbor` can be confusing to new users, especially because of the case you mentioned here.
But because we use `neighbors` as keywords and in our examples we use `neighbors` instead of `vectors`, I feel like `neighbors` is more consistent, or at least we will have to use them interchangeably.
Since we added the example distinguishing between `env_neighbors` and `neighbors`, and extra doc in their definition to explain the different, I think it is pretty clear.

Also `neighbors` actually makes more sense once one understands how it works, because they are indeed neighbors, just used differently for this computation. Vectors are implied, and used only to specify the location of the neighbors.

Reply:
Rather than debating this amongst ourselves, I think the best approach here is to solicit opinions from people who aren't familiar with the code. @Charlottez112 could you ask a couple of Glotzer lab members who don't currently use environment matching to read the docs and explain what they think the docs mean? That will be much more useful than our opinions since at this point everyone on this thread already has a solid understanding of the code.

Reply:
I agree with @vyasr that soliciting people's opinions will be the way to go- we can also change the "neighbors" argument in EnvironmentMatchMotif to "env_neighbors" if that would be more consistent, because you're completely right that they serve the same function. For historical reasons those terms weren't consistent but perhaps it would be better to make them consistent now that we are clarifying things. Perhaps then the use of "neighbors" (or "env_neighbors") wouldn't throw people off with respect to the neighbors vs vectors debate. I think either term would honestly probably be fine as long as we are clear in our explanation of terms. Describing motifs as sets of "vectors" in EnvironmentMotifMatch appealed to me because the motif that the user provides and might want to search for within the system can be any set of vectors, not necessarily an environment "neighborhood" already found via EnvironmentCluster. But again, I could really go either way.